src/Sylius/Bundle/CoreBundle/Cart/ItemResolver.php

```diff
@@ -60,27 +61,35 @@ class ItemResolver implements ItemResolverInterface
     protected $channelContext;
 
     /**
-     * @param CartProviderInterface          $cartProvider
-     * @param RepositoryInterface            $productRepository
-     * @param FormFactoryInterface           $formFactory
-     * @param AvailabilityCheckerInterface   $availabilityChecker
-     * @param DelegatingCalculatorInterface  $priceCalculator
-     * @param ChannelContextInterface        $channelContext
+     * @var VariantResolverInterface
+     */
+    protected $defaultVariantResolver;
+
+    /**
+     * @param CartProviderInterface $cartProvider
+     * @param RepositoryInterface $productRepository
+     * @param FormFactoryInterface $formFactory
+     * @param AvailabilityCheckerInterface $availabilityChecker
+     * @param DelegatingCalculatorInterface $priceCalculator
+     * @param ChannelContextInterface $channelContext
+     * @param VariantResolverInterface $defaultVariantResolver
      */
     public function __construct(
-        CartProviderInterface          $cartProvider,
-        RepositoryInterface            $productRepository,
-        FormFactoryInterface           $formFactory,
-        AvailabilityCheckerInterface   $availabilityChecker,
-        DelegatingCalculatorInterface  $priceCalculator,
-        ChannelContextInterface        $channelContext
+        CartProviderInterface $cartProvider,
+        RepositoryInterface $productRepository,
+        FormFactoryInterface $formFactory,
+        AvailabilityCheckerInterface $availabilityChecker,
+        DelegatingCalculatorInterface $priceCalculator,
+        ChannelContextInterface $channelContext,
+        VariantResolverInterface $defaultVariantResolver
```

Comment:
7 injected services? We should consider some refactoring and extract some of this logic... It starts to be too complicated.